tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,289 @@
+import shlex
+
+import pytest
+from kubernetes.dynamic.exceptions import NotFoundError
+from ocp_resources.exceptions import ExecOnPodError
+from ocp_resources.network_policy import NetworkPolicy
+from ocp_resources.pod import Pod
+from ocp_resources.service import Service
+
+from utilities.constants import (
+    KUBEVIRT_APISERVER_PROXY_NP,
+    KUBEVIRT_CONSOLE_PLUGIN_NP,
+    POD_CONTAINER_SPEC,
+    POD_SECURITY_CONTEXT_SPEC,
+)
+
+pytestmark = [pytest.mark.post_upgrade, pytest.mark.sno, pytest.mark.arm64, pytest.mark.s390x]
+
+TEST_SERVER_PORT = 9876
+TEST_SERVER_APP_LABEL = "network-policy-server"
+
+
+class AllowAllNetworkPolicy(NetworkPolicy):
+    def __init__(self, name, namespace, client, match_labels):
+        super().__init__(name=name, namespace=namespace, client=client)
+        self.match_labels = match_labels
+
+    def to_dict(self):
+        super().to_dict()
+        self.res["spec"] = {
+            "podSelector": {"matchLabels": self.match_labels},
+            "policyTypes": ["Ingress", "Egress"],
+            "ingress": [{}],
+            "egress": [{}],
+        }
+
+
+class ServerPod(Pod):
+    def __init__(self, name, namespace, client, security_context, containers):
+        super().__init__(
+            name=name, namespace=namespace, client=client, security_context=security_context, containers=containers
+        )
+
+    def to_dict(self):
+        super().to_dict()
+        self.res.setdefault("metadata", {}).setdefault("labels", {}).update({"app": TEST_SERVER_APP_LABEL})
+
+
+@pytest.fixture
+def network_policy_by_name(admin_client, hco_namespace, network_policy_name):
+    try:
+        network_policies = list(
+            NetworkPolicy.get(dyn_client=admin_client, namespace=hco_namespace.name, name=network_policy_name)
+        )
+        assert len(network_policies) == 1, (
+            f"Expected exactly 1 NetworkPolicy {network_policy_name} in namespace {hco_namespace.name}, "
+            f"found {len(network_policies)}"
+        )
```

Comment:
not sure I understand the check here.
`get` is used with a name and namespace - how can there be more than 1?

to make sure the expected NetworkPolicy exists you can:

```
return NetworkPolicy(client=admin_client, namespace=hco_namespace.name, name=network_policy_name, ensure_exists=True)
        )
```